field/src/field.rs

```diff
@@ -363,6 +363,16 @@ pub trait PrimeField64: PrimeField {
 
     /// Return the representative of `value` that is less than `ORDER_U64`.
     fn as_canonical_u64(&self) -> u64;
+
+    /// Convert a field element to a `u64` such that any two field elements
+    /// are converted to the same `u64` if and only if they represent the same value.
```

Comment:
nit: maybe briefly clarify that this should hold even across different targets, in other words its behavior should be portable